lib/beaker/subcommand.rb

```diff
@@ -6,15 +6,41 @@ module Beaker
   class Subcommand < Thor
     SubcommandUtil = Beaker::Subcommands::SubcommandUtil
 
+    def initialize(*args)
+      super
+      @@config = SubcommandUtil.load_config()
+    end
+
     desc "init", "Initialises the beaker test environment configuration"
     option :hypervisor, :type => :string, :required => true
     long_desc <<-LONGDESC
-      Initialises a beaker environment configuration
+    Initialises a beaker environment configuration
     LONGDESC
     def init()
       SubcommandUtil.verify_init_args(options)
       SubcommandUtil.require_tasks()
       SubcommandUtil.init_hypervisor(options)
+      SubcommandUtil.store_config(options)
+      SubcommandUtil.delete_config([:provisioned, :hosts])
+    end
+
+    desc "provision", "Provisions the beaker test environment"
+    option :force, :type => :boolean
+    option :validate, :type => :boolean, :default => true
+    option :configure, :type => :boolean, :default => true
+    long_desc <<-LONGDESC
+    Provisions a beaker environment
+    LONGDESC
+    def provision()
+      provisioned = @@config.transaction { @@config[:provisioned] }
+
+      if !provisioned or options[:force]
+        hypervisor = @@config.transaction { @@config[:hypervisor] }
+        hosts = SubcommandUtil.init_provision(hypervisor, options)
+        SubcommandUtil.store_config({:provisioned => true, :hosts => hosts})
```

Comment:
So retrieving and writing out the host info is actually pretty pointless since I can just call preserve_hosts_file directly. I'll update the code to do that instead